rfcs/0005-pablo-distribution.md

```diff
@@ -0,0 +1,667 @@
+
+<img src="0005-pablo-distribution-assets/images/stem-35912508e8bf41c1a7f94b93abcec3aa.png" width="98" height="19" alt="stem 35912508e8bf41c1a7f94b93abcec3aa" /></span>
+
+trading fees received <img src="0005-pablo-distribution-assets/images/stem-acbc3160f2b6a5977e6ac719418e0581.png" width="106" height="19" alt="stem acbc3160f2b6a5977e6ac719418e0581" /></span>
+
+"When pool size <img src="0005-pablo-distribution-assets/images/stem-c8165429df4fe2a9cc08c1a6949ead7c.png" width="30" height="12" alt="stem c8165429df4fe2a9cc08c1a6949ead7c" /></span>
+increases the amount of trading fees received <img src="0005-pablo-distribution-assets/images/stem-332cc365a4987aacce0ead01b8bdcc0b.png" width="9" height="6" alt="stem 332cc365a4987aacce0ead01b8bdcc0b" /></span>
+reduces for a particular LP position.
+
+For large pool sizes of <img src="0005-pablo-distribution-assets/images/stem-c8165429df4fe2a9cc08c1a6949ead7c.png" width="30" height="12" alt="stem c8165429df4fe2a9cc08c1a6949ead7c" /></span>
+(steady state) this effect is negligible, hence it’s a good enough
+strategy to distribute fees.
+
+But if required this effect can be negated by increasing the trading fee
+by a <img src="0005-pablo-distribution-assets/images/stem-e64be84a4eef601683d61de156018075.png" width="24" height="10" alt="stem e64be84a4eef601683d61de156018075" /></span>
+while at the same time subtracting it from the total fees paid out
+already to liquidity providers. Refer [Trading Fee Inflation to Avoid
+Dilution of LPs](#_trading_fee_inflation_to_avoid_dilution_of_lps).
+
+#### 5.2.3. PBLO Staker Trading Fee Distribution
+
+This is the reward a `PBLO` staker receives from the trading fees of
+Pablo pools. It is equal to the protocol fee charged on Pablo pools.
+This can be accomplished by calling the already existing
+`StakingReward.transfer_reward` interface as follows. According to
+product there is also a need to convert whatever the fee asset in to
+PBLO to create a demand/additional value for PBLO.
+
+<img src="0005-pablo-distribution-assets/images/images/pablo-fNFT-pblo-staking-fee-distro.png" width="510" height="346" alt="pablo fNFT pblo staking fee distro" />
+
+Will it need a change in
+[this](https://github.com/ComposableFi/composable/blob/main/frame/composable-traits/src/staking_rewards.rs#L96)
+?
+
+### 5.3. Pallet Staking Rewards - LP/PICA/PBLO/Other Token Staking Reward Pools
+
+This section covers how the staking rewards are distributed using the
+[staking rewards
+pallet](https://github.com/ComposableFi/composable/tree/main/frame/staking-rewards).
+
+#### 5.3.1. Analysis of Reward Calculations
+
+In order to create the necessary reward pool as well as the rewarding
+rate for stakers the following model can be used. It tries to address
+the following constraints (TODO move to requirements)
+
+1.  Allow <span id="rate">specification of the reward rate for a
```

Comment:
dynamically changing rate [postponed]

Reply:
It is not, it will be implemented with the first iteration, refer section 5.3.7